webapp-src/src/Login/GrantScope.js

```diff
@@ -89,10 +89,20 @@ class GrantScope extends Component {
       infoSomeScopeUnavailable = <div className="alert alert-info" role="alert">{i18next.t("login.info-some-scope-unavailable")}</div>
     }
     return (
+    (!Array.isArray(this.state.scope) || this.state.scope.length < 1)
```

Comment:
I think `Array.isArray` is overkill, `this.state.scope` is always supposed to be an array, even an empty one.

Also, using a one-liner makes it more difficult to read than a simple `if`

Reply:
You know, I vaguely remember having a simple `if (this.state.scope === [])` conditional there but it would never be satisfied. I was working under time pressure, though, so I might be wrong. I'd just suggest that you test if you replace with a simpler statement.

As for tertiary conditionals, I confess I have a penchant for them. Sorry… :)

Reply:
> As for tertiary conditionals, I confess I have a penchant for them. Sorry… :)

I know the feeling, I use them a lot in the backend

Reply:
@yrammos , I realize I already made something similar, but in the [App.js](https://github.com/babelouest/glewlwyd/blob/master/webapp-src/src/Login/App.js#L271)
Although, there's only the error message displayed, I think we should add the connected user as well. The error message is in the i18n entry `login.error-scope-unavailable`

Reply:
@babelouest I don’t think I came across that ,but don’t hesitate to tell me if you’d like me to rewire something.

Reply:
For now, I reset to the original code, I made it specifically when a user is redirected to the login page with a scope list and the user has no scope from the list. Can you retry your tests that lead you to your similar change? I may have forgotten some use cases. If so I'll fix the original code.

Reply:
@babelouest ah, I now see what you're referring to. The original code indeed seems to work just as well.